src/Microsoft.ML.Data/Training/TrainerEstimatorBase.cs

```diff
@@ -122,8 +120,8 @@ private void CheckInputSchema(SchemaShape inputSchema)
 
         protected virtual void CheckLabelCompatible(SchemaShape.Column labelCol)
         {
-            Contracts.CheckValue(labelCol, nameof(labelCol));
-            Contracts.AssertValue(LabelColumn);
+            Contracts.Check(labelCol.IsValid, nameof(labelCol));
```

Comment:
>Contracts.Check(labelCol.IsValid, nameof(labelCol)); [](start = 12, length = 52)

Host.CheckParam(labelCol.IsValid, nameof(labelCol), "not initialized properly") ?